`DEMO_MODE=1 make run-server` works where as `DEMO_MODE=true make run-server` fails

when I run `DEMO_MODE=true make run-server`  I get this error,

```
Failed to parse arguments as literal values: 'app("info", true)'
[2024-05-14 12:44:31 -0400] [69176] [INFO] Worker exiting (pid: 69176)
[2024-05-14 12:44:32 -0400] [69172] [ERROR] Worker (pid:69179) exited with code 4
```

`DEMO_MODE=1 make run-server` and `DEMO_MODE=True make run-server` works fine

--- a/Makefile
+++ b/Makefile
@@ -10,9 +10,8 @@ DROP_TABLES ?= False
 run-postgres:
 	$(CONTAINER_RUNTIME) run -it -v data:/var/lib/postgresql/data -e POSTGRES_USER=kai -e POSTGRES_PASSWORD=dog8code -e POSTGRES_DB=kai -p 5432:5432 docker.io/pgvector/pgvector:pg15
 
-# Can adjust logging level by defining env var: KAI_LOG_LEVEL
 run-server:
-	PYTHONPATH=$(KAI_PYTHON_PATH) gunicorn --timeout 3600 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app("$(LOGLEVEL)", "$(DEMO_MODE)")'
+	PYTHONPATH=$(KAI_PYTHON_PATH) LOGLEVEL=$(LOGLEVEL) DEMO_MODE=$(DEMO_MODE) gunicorn --timeout 3600 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app()'
 
 load-data:
 	PYTHONPATH=$(KAI_PYTHON_PATH) python ./kai/service/incident_store/psql.py  --config_filepath ./kai/config.toml --drop_tables $(DROP_TABLES)

--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ run-postgres:
 
 # Can adjust logging level by defining env var: KAI_LOG_LEVEL
 run-server:
-	PYTHONPATH=$(KAI_PYTHON_PATH) gunicorn --timeout 3600 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app("$(LOGLEVEL)", $(DEMO_MODE))'
+	PYTHONPATH=$(KAI_PYTHON_PATH) gunicorn --timeout 3600 -w $(NUM_WORKERS) --bind localhost:8080 --worker-class aiohttp.GunicornWebWorker 'kai.server:app("$(LOGLEVEL)", "$(DEMO_MODE)")'
 
 load-data:
 	PYTHONPATH=$(KAI_PYTHON_PATH) python ./kai/service/incident_store/psql.py  --config_filepath ./kai/config.toml --drop_tables $(DROP_TABLES)

--- a/kai/server.py
+++ b/kai/server.py
@@ -240,7 +240,10 @@ async def get_incident_solutions_for_file(request: Request):
     return web.json_response(result)
 
 
-def app(log_level: Optional[str] = None, demo_mode: Optional[bool] = None):
+def app(log_level: Optional[str] = None, demo_mode: Optional[bool | str] = None):
+    if isinstance(demo_mode, str):
+        demo_mode = demo_mode.lower() == "true"
+
     webapp = web.Application()
 
     config: KaiConfig

--- a/kai/server.py
+++ b/kai/server.py
@@ -240,10 +240,7 @@ async def get_incident_solutions_for_file(request: Request):
     return web.json_response(result)
 
 
-def app(log_level: Optional[str] = None, demo_mode: Optional[bool | str] = None):
-    if isinstance(demo_mode, str):
-        demo_mode = demo_mode.lower() == "true"
-
+def app():
     webapp = web.Application()
 
     config: KaiConfig
@@ -260,10 +257,10 @@ def app(log_level: Optional[str] = None, demo_mode: Optional[bool | str] = None)
     else:
         raise FileNotFoundError("Config file not found.")
 
-    if log_level:
-        config.log_level = log_level
-    if demo_mode:
-        config.demo_mode = demo_mode
+    if os.getenv("LOGLEVEL") is not None:
+        config.log_level = os.getenv("LOGLEVEL").upper()
+    if os.getenv("DEMO_MODE") is not None:
+        config.demo_mode = os.getenv("DEMO_MODE").lower() == "true"
 
     print(f"Config loaded: {pprint.pformat(config)}")
 
@@ -319,7 +316,10 @@ Example: --loglevel debug (default: warning)""",
 
     args, _ = arg_parser.parse_known_args()
 
-    web.run_app(app(args.loglevel, args.demo_mode))
+    os.environ["LOGLEVEL"] = str(args.loglevel)
+    os.environ["DEMO_MODE"] = str(args.demo_mode).lower()
+
+    web.run_app(app())
 
 
 if __name__ == "__main__":

